core/src/main/java/org/opensearch/sql/analysis/ExpressionAnalyzer.java

```diff
@@ -193,9 +193,8 @@ public Expression visitWindowFunction(WindowFunction node, AnalysisContext conte
 
   @Override
   public Expression visitHighlightFunction(HighlightFunction node, AnalysisContext context) {
-    Expression expr = node.getHighlightField().accept(this, context);
-    String highlightStr = "highlight(" + StringUtils.unquoteText(expr.toString()) + ")";
-    return new ReferenceExpression(highlightStr, ExprCoreType.STRING);
+    return (node.getName() != null) ? new ReferenceExpression(node.getName(), ExprCoreType.STRING)
```

Comment:
When is `node.getName` null?

Reply:
Not possible unless new code is integrated in which the variable is not set.